From a7d662bda95c33131126b2ad45bbb073d93b85d3 Mon Sep 17 00:00:00 2001
From: Christopher Faulet <cfaulet@haproxy.com>
Date: Thu, 5 Jan 2023 14:32:33 +0100
Subject: [PATCH] BUG/MINOR: cfgparse: Use the right proxy list during
 validity check of sinks

The commit 7af321580 ("BUG/MEDIUM: sink: bad init sequence on tcp sink from
a ring.") was not properly backported. The main loop on the proxy list in
check_config_validity() is not initialized with the right list when validity
checks are performed on sinks. It leads to errors on startup when a ring is
configured.

It is a 2.2-specific commit. Thus there is no upstream ID. No backport
needed.
---
 src/cfgparse.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 100a0e8..f07d36d 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -2283,7 +2283,7 @@ int check_config_validity()
 	init_proxies_list = proxies_list;
 
 init_proxies_list_stage1:
-	for (curproxy = proxies_list; curproxy; curproxy = curproxy->next) {
+	for (curproxy = init_proxies_list; curproxy; curproxy = curproxy->next) {
 		struct switching_rule *rule;
 		struct server_rule *srule;
 		struct sticking_rule *mrule;
-- 
1.7.10.4

